### PR TITLE
Feature to call controller action server to follow path

### PR DESCRIPTION
**Pull request by Ekanshh**
---

## Basic Info

| Info | Please fill out this column |
| ------ | ----------- |
| Ticket(s) this addresses   | #2415  |
| Primary OS tested on | Ubuntu|
| Robotic platform tested on | Gazebo simulation of turtlebot3_waffle |

---

## Description of contribution in a few bullet points

* I added this new method `followPath` in robot_navigator.py  to call the controller with a given path
* I created a new example `follow_path_example_launch.py` to highlight this new API function.
* I updated the relevant README.md section to highlight this new feature.

---

#### For Maintainers: 
- [x] Check that any new parameters added are updated in navigation.ros.org
- [x] Check that any significant change is added to the migration guide
- [x] Check that any new features **OR** changes to existing behaviors are reflected in the tuning guide
- [x] Check that any new functions have Doxygen added
- [x] Check that any new features have test coverage
- [x] Check that any new plugins is added to the plugins page
- [x] If BT Node, Additionally: add to BT's XML index of nodes for groot, BT package's readme table, and BT library lists

### CODE REVIEW

File: nav2_simple_commander/README.md [hunk 1]
```diff
@@ -20,6 +20,7 @@ The methods provided by the basic navigator are shown below, with inputs and exp
 | goThroughPoses(poses)             | Requests the robot to drive through a set of poses (list of `PoseStamped`).|
 | goToPose(pose)                    | Requests the robot to drive to a pose (`PoseStamped`).                     |
 | followWaypoints(poses)            | Requests the robot to follow a set of waypoints (list of `PoseStamped`). This will execute the specific `TaskExecutor` at each pose.   |
+| followPath(path)                  | Requests the robot to follow a path from a starting to a goal `PoseStamped`, `nav_msgs/Path`.     |
```
Comment (SteveMacenski) [inline nav2_simple_commander/README.md:23]: https://navigation.ros.org/commander_api/index.html Add to this API too!

File: nav2_simple_commander/README.md [hunk 2]
```diff
@@ -102,8 +103,8 @@ The `nav2_simple_commander` has a few examples to highlight the API functions av
 
 - `example_nav_to_pose.py` - Demonstrates the navigate to pose capabilities of the navigator, as well as a number of auxiliary methods.
 - `example_nav_through_poses.py` - Demonstrates the navigate through poses capabilities of the navigator, as well as a number of auxiliary methods.
-- `example_waypoint_follower.py` - Demonstrates the waypoint following capabilities of the navigator, as well as a number of auxiliary methods required.
-
+- `example_waypoint_follower.py` - Demonstrates the waypoint following capabilities of the navigator, as well as a number of auxiliary methods.
+- `example_follow_path.py` - Demonstrates the path following capabilities of the navigator, as well as a number of auxiliary methods.
```
Comment (SteveMacenski) [inline nav2_simple_commander/README.md:107]: Ditto on the nav2 website page

File: nav2_simple_commander/nav2_simple_commander/robot_navigator.py [hunk 3]
```diff
@@ -231,6 +256,8 @@ def getPath(self, start, goal):
             return None
 
         return self.result_future.result().result.path
+
```
Comment (SteveMacenski) [inline nav2_simple_commander/nav2_simple_commander/robot_navigator.py:259]: run `ament_flake8` locally and fix all the issues it brings up. CI was complaining about some styling inconsistencies

**Review by SteveMacenski — CHANGES_REQUESTED** [review]
LGTM, just a few small updates just to polish it up!

File: nav2_simple_commander/README.md (hunk 1 above)
Comment (Ekanshh) [reply nav2_simple_commander/README.md:23]: Updated the required changes at this PR https://github.com/ros-planning/navigation.ros.org/pull/272

File: nav2_simple_commander/README.md (hunk 2 above)
Comment (Ekanshh) [reply nav2_simple_commander/README.md:107]: Updated the required changes at this PR https://github.com/ros-planning/navigation.ros.org/pull/272

File: nav2_simple_commander/nav2_simple_commander/robot_navigator.py (hunk 3 above)
Comment (Ekanshh) [reply nav2_simple_commander/nav2_simple_commander/robot_navigator.py]: Fixed it in commit b874c9a65d81679ccac08327565e355c8a0178b4. Now CI is failing for some other reason, please have a look at it.

**Review by SteveMacenski — APPROVED** [review]

**Review by SteveMacenski — APPROVED** [review]

**Comment by SteveMacenski** [comment]
Awesome job and I appreciate it! 

Any interest in expanding on this with the item from the ticket ` Add default arguments for full action API to functions (that don't necessarily need to be provided, but if they are, use them. example: behavior tree xmls for nav requests)`? 

Essentially the action definitions https://github.com/ros-planning/navigation2/tree/main/nav2_msgs/action contain more than just a `path` field for follow path (and others). The task is to change the `followPath()` function to take in the other parameters optionally so they can be specified if a user wants to. 

So the arguments should default to `None` and checked in the function to populate the field in the request if not `None`. That way the simple API you made can be used, but additional fields for the full action API can be leveraged by users that want more control.

This is true of most of the actions with the python API above

**Comment by Ekanshh** [comment]
> Awesome job and I appreciate it!
> 
> Any interest in expanding on this with the item from the ticket ` Add default arguments for full action API to functions (that don't necessarily need to be provided, but if they are, use them. example: behavior tree xmls for nav requests)`?
> 
> Essentially the action definitions https://github.com/ros-planning/navigation2/tree/main/nav2_msgs/action contain more than just a `path` field for follow path (and others). The task is to change the `followPath()` function to take in the other parameters optionally so they can be specified if a user wants to.
> 
> So the arguments should default to `None` and checked in the function to populate the field in the request if not `None`. That way the simple API you made can be used, but additional fields for the full action API can be leveraged by users that want more control.
> 
> This is true of most of the actions with the python API above

Sure. I can work on it, starting from next week. :)

**Pull request merged**